### ISSUE TITLE
Minimize shortcut in Firefox on macOS doesn't work

### ISSUE THREAD

**Issue by sometoby** [thelounge/thelounge#2442]
On macOS, whichever shortcut The Lounge binds to ⌘M obscures Firefox' native minimize shortcut, making it impossible to minimize Firefox using ⌘M when a tab with The Lounge is active.

**Comment by Jay2k1** [comment]
> whichever shortcut The Lounge binds to ⌘M

![image](https://user-images.githubusercontent.com/2070503/40117026-f42f953e-5915-11e8-9a87-0c009545c985.png)

It's not only Firefox - Chrome, Vivaldi and Safari are affected too, since ⌘M is the default shortcut for "minimize to Dock" on MacOS.

**Comment by sometoby** [comment]
I'm unable to reproduce the problem in Safari.

**Comment by Jay2k1** [comment]
Oh, indeed. I didn't actually test the behaviour, I just checked each browsers' "Window" menu.

![image](https://user-images.githubusercontent.com/2070503/40117307-e03dae5c-5916-11e8-8031-7118c53f7e63.png)

**Comment by MaxLeiter** [comment]
This only occurs when the message input box isn't focused. Not too sure we can do anything here except change the keybind. As of #2215

**Comment by MaxLeiter** [comment]
No, I mean I expected keybinds like bold and monospace to focus on the input box. If that’s not the actual behavior then I can’t reproduce this bug

**Comment by xPaw** [comment]
Lounge's keybinds don't do anything (not even set focus) when input is focused. This means any OS/browser keybiding should work just fine if input isn't focus.

**Comment by xPaw** [comment]
Is there anything we can/should do here?

As of now, if you don't have the input focused, TL's keybinds will not work, and the browser's one should.

**Comment by Jay2k1** [comment]
OP's issue is that they want to minimize Firefox with cmd+M and TL inserts the monospace control char instead. This can be avoided by not having the input focused. I'd call this a minor annoyance.

What's worse is that in Safari, cmd+M *always* minimizes the window, even with input focused, so you cannot make monospace formatted text in Safari as of now. This should be fixed if possible, I just doubt that there's anything we can do except change keybinds, which should probably be avoided...

**Comment by xPaw** [comment]
>This can be avoided by not having the input focused. I'd call this a minor annoyance.

Agreed.

> What's worse is that in Safari, cmd+M always minimizes the window, even with input focused

Can we add a keybind just for safari, like cmd+shift+M or something?

**Comment by Jay2k1** [comment]
Sounds reasonable. Cmd+O always brings up "file open" dialog too, so that needs a shift modifier too.
Cmd+(B|U|I|S) do work, but I suggest adding the shift modifier to all of them, not just some. The question is, do we do this for all mac binds in general or can we somehow make a safari-only solution - and even if, should we? Should a Mac user need to use different shortcuts based on the browser they use?
As a (part-time) Mac user myself, I don't like the idea of suddenly not being able to use cmd+key shortcuts anymore, but that's probably something I'd get used to. The most pleasant solution is probably to _also_ allow shift-modified shortcuts, so non-Safari users get to keep their shortcuts and Safari users have a solution. This would just raise the question what to display on the help page. Maybe a note ("Safari users need to add shift"?) or we can maybe show different shortcuts based not only on OS but also the browser?

**Comment by xPaw** [comment]
> The most pleasant solution is probably to also allow shift-modified shortcut

That's what I suggested, _add_ cmd+shift modifiers.

> or we can maybe show different shortcuts based not only on OS but also the browser?

We already have `is-apple` which switches Ctrl to Cmd for Mac users.

**Comment by Jay2k1** [comment]
> We already have `is-apple` which switches Ctrl to Cmd for Mac users.

I know, hence asking if we can do it not only for the different OS but also for a specific browser...
